### PR TITLE
feat: impl runtime canister stable storage

### PR DESCRIPTION
**Pull request by ozwaldorf**

### CODE REVIEW

File: ic-kit-runtime/src/canister.rs [hunk 1]
```diff
@@ -1021,35 +1021,64 @@ impl Ic0CallHandlerProxy for Canister {
...
+            .unwrap_or(i32::max_value()))
     }
 
     fn stable_grow(&mut self, new_pages: i32) -> Result<i32, String> {
-        todo!()
+        let size: i32 = self.stable.stable_size() as i32;
+        let max_size = i32::max_value();
+
+        if size + new_pages > max_size {
+            // do we return -1 ?
```
Comment (qti3e) [inline ic-kit-runtime/src/canister.rs]: Yes

File: ic-kit-runtime/src/canister.rs [hunk 2]
```diff
@@ -1021,35 +1021,64 @@ impl Ic0CallHandlerProxy for Canister {
     }
 
     fn stable_size(&mut self) -> Result<i32, String> {
-        todo!()
+        Ok(self
+            .stable
+            .stable_size()
+            .try_into()
+            .unwrap_or(i32::max_value()))
```
Comment (qti3e) [inline ic-kit-runtime/src/canister.rs]: It should return an error if it can't fit in 32bit

**Review by qti3e — CHANGES_REQUESTED** [review]

File: ic-kit-runtime/src/canister.rs (hunk 2 above)
Comment (ozwaldorf) [reply ic-kit-runtime/src/canister.rs]: 2a2e6a8

File: ic-kit-runtime/src/canister.rs (hunk 1 above)
Comment (ozwaldorf) [reply ic-kit-runtime/src/canister.rs]: 2a2e6a8

**Review by qti3e — APPROVED** [review]

**Pull request merged**